com.unity.ml-agents/Runtime/Sensors/IOverlapChecker.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using UnityEngine;
+
+namespace Unity.MLAgents.Sensors
+{
+    internal interface IOverlapChecker
```

Comment:
Could this name be made a bit more generic?  You may not want to just do overlap checks, you may want to do ray casts, or box casts, etc. 

Does `IGridPerception` make sense?

Reply:
Also, please add a class level comment.